### PR TITLE
Integrate airtap

### PR DESCRIPTION
**Pull request by mcollina**
Supersedes #340 because of the elevated permission to run Sauce Labs with travis

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +25,8 @@
   "scripts": {
     "test": "tap -j 4 test/parallel/*.js test/ours/*.js",
     "ci": "TAP=1 tap test/parallel/*.js test/ours/*.js | tee test.tap",
+    "test-browsers": "airtap --sauce-connect --no-coverage -- test/browser.js",
```
Comment (vweevers) [inline package.json]: With latest `airtap` (0.0.9) you no longer need `--no-coverage`. Collecting code coverage became opt-in.

**Comment by mcollina** [comment]
I'm getting:

```
Error:
Airtap tried to run tests in Sauce Labs, however no credentials were provided.
See doc/cloud-testing.md for info on how to setup cloud testing.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! readable-stream@3.0.0-dev0 test-browsers: `airtap --sauce-connect -- test/browser.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the readable-stream@3.0.0-dev0 test-browsers script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2018-07-05T09_32_02_611Z-debug.log
```

have you got any clues? cc @vweevers @ralphtheninja

File: .travis.yml [hunk 2]
```diff
@@ -18,9 +18,9 @@ matrix:
...
       sauce_connect: true
       hosts:
-        - airtap.local
+      - airtap.local
 script: npm run $CMD
 env:
   global:
-  - secure: rE2Vvo7vnjabYNULNyLFxOyt98BoJexDqsiOnfiD6kLYYsiQGfr/sbZkPMOFm9qfQG7pjqx+zZWZjGSswhTt+626C0t/njXqug7Yps4c3dFblzGfreQHp7wNX5TFsvrxd6dAowVasMp61sJcRnB2w8cUzoe3RAYUDHyiHktwqMc=
-  - secure: g9YINaKAdMatsJ28G9jCGbSaguXCyxSTy+pBO6Ch0Cf57ZLOTka3HqDj8p3nV28LUIHZ3ut5WO43CeYKwt4AUtLpBS3a0dndHdY6D83uY6b2qh5hXlrcbeQTq2cvw2y95F7hm4D1kwrgZ7ViqaKggRcEupAL69YbJnxeUDKWEdI=
+  - secure: nk/kDUZV385ey9kkrGi6EHvnslTAW+GEHGrjbFIsb8Foq01bL9oj0uPaVc5kaVWCHst3WsIdVMA1Hos/lN/WMHRyfOiSrVVWv0sVnzg0YaM9cBbn5jwXn30Z645REgQcoEDKrYFp8rE3RpPFgYDFWNh948qwPtwtnwNhoq/Gd9A=
```
Comment (ralphtheninja) [inline .travis.yml]: @mcollina These are different from https://github.com/nodejs/readable-stream/pull/340/commits/40bd7adc039db9e494648c4200ed5cb6973f4c1e which worked?

Comment (ralphtheninja) [reply .travis.yml]: Oh, you might need to escape the `/` there

screenshot from 2018-07-05 12-16-11

**Comment by ralphtheninja** [comment]
From airtap:

```js
} else if (!config.username || !config.key) {                                                                                                                                                                      
  console.error(chalk.red('Error:'))                                                                                                                                                                               
  console.error(chalk.red('Airtap tried to run tests in Sauce Labs, however no credentials were provided.'))                                                                                                       
  console.error(chalk.cyan('See doc/cloud-testing.md for info on how to setup cloud testing.'))                                                                                                                    
  process.exit(1)                                                                                                                                                                                                  
}
```

And `username` and `key` are set as:

```js
config.username = process.env.SAUCE_USERNAME || config.sauce_username                                                                                                                                              
config.key = process.env.SAUCE_ACCESS_KEY || config.sauce_key                                                                                                                                                      
```

@mcollina To which environment variables are you setting the secure values?

**Comment by ralphtheninja** [comment]
@vweevers I'm on my way to Amsterdam and have to pack so don't have more time atm. Maybe you can jump in a bit?

**Comment by mcollina** [comment]
SAUCE_USERNAME and SAUCE_ACCESS_KEY.

But here is what I'm seeing on travis-ci.com:

https://travis-ci.com/nodejs/readable-stream/jobs/132925702#L430-L431

**Comment by vweevers** [comment]
Maybe try defining the vars in Travis instead?

image

**Comment by mcollina** [comment]
Setting in on travis worked!!!

**Comment by mcollina** [comment]
Works on IE 11 as well. Hurray!

**Pull request merged**